centaur/src/main/scala/centaur/test/Test.scala

```diff
@@ -560,19 +560,40 @@ object Operations extends StrictLogging {
       lazy val inExpectedButNotInActual = expected.diff(actual)
 
       if (!workflow.allowOtherOutputs && inActualButNotInExpected.nonEmpty) {
-        val message =
-          s"In actual outputs but not in expected and other outputs not allowed: ${inActualButNotInExpected.mkString(", ")}"
-        IO.raiseError(CentaurTestException(message, workflow, submittedWorkflow))
+        if (!checkIfActuallySame(inActualButNotInExpected, inExpectedButNotInActual)) {
+          val message =
+            s"In actual outputs but not in expected and other outputs not allowed: ${inActualButNotInExpected.mkString(", ")}"
+          IO.raiseError(CentaurTestException(message, workflow, submittedWorkflow))
+        } else {
+          IO.unit
+        }
       } else if (inExpectedButNotInActual.nonEmpty) {
-        val message =
-          s"In actual outputs but not in expected: ${inExpectedButNotInActual.mkString(", ")}" + System.lineSeparator +
-            s"In expected outputs but not in actual: ${inExpectedButNotInActual.mkString(", ")}"
-        IO.raiseError(CentaurTestException(message, workflow, submittedWorkflow))
+        if (!checkIfActuallySame(inExpectedButNotInActual, inActualButNotInExpected)) {
+          val message =
+            s"In actual outputs but not in expected: ${inExpectedButNotInActual.mkString(", ")}" + System.lineSeparator +
+              s"In expected outputs but not in actual: ${inExpectedButNotInActual.mkString(", ")}"
+          IO.raiseError(CentaurTestException(message, workflow, submittedWorkflow))
+        } else {
+          IO.unit
+        }
       } else {
         IO.unit
       }
     }
 
+    private def checkIfActuallySame(firstSet: Set[(String, JsValue)], secondSet: Set[(String, JsValue)]): Boolean = {
```

Comment:
Naming: this function alludes to a "kinda same"/"actually same" duality... not sure that's what's happening

Reply:
Essentially the actual outputs are in a "cleaned" state where extraneous quotation marks have been removed, but the expected outputs might have extra quotation marks if the output is a String (e.g., the actual output will be "hello" but the expected output will read "\"hello\""). This function just cleans the outputs to make sure that we're not reporting a false difference caused by quotes that should have been removed.

Reply:
Could you add a comment to that affect?